src/targets/gpu/include/migraphx/gpu/gemm.hpp

```diff
@@ -132,6 +133,43 @@ struct rocblas_gemm
     {
         return shapes.size() - 1;
     }
+
+    void finalize(context& ctx, const shape& output_shape, const std::vector<shape>& input_shapes)
+    {
+        rocblas_initialize();
+        rocblas_status_ check_valid;
+        if(ctx.get_exhaustive_tune_flag() && solution_idx == 0)
+        {
+            if(this->name() == "gpu::gemm")
+            {
+                auto gemmImpl = gemm_impl<float>(
+                    output_shape, input_shapes, alpha, beta, int8_x4_format, compute_fp32);
+                solution_idx = gemmImpl.tune(ctx, input_shapes);
+                check_valid  = gemmImpl.validate(ctx, input_shapes, solution_idx);
+            }
+            else
+            {
+                auto gemmImpl = gemm_impl<int32_t>(
+                    output_shape, input_shapes, alpha, beta, int8_x4_format, compute_fp32);
+                solution_idx = gemmImpl.tune(ctx, input_shapes);
+                check_valid  = gemmImpl.validate(ctx, input_shapes, solution_idx);
+            }
+        }
+        else if(solution_idx != 0)
+        {
+            auto gemmImpl = gemm_impl<int32_t>(
```

Comment:
This should check first if it is `quant_gemm` or regular gemm . If it quant-Gemm then only use `int32_t` otherwise use `float`.

Reply:
Is that exactly the same as the if-else block above, then?  The other kinds of gemm I see defined in README.MD are gpu::int8_gemm_pack_a and gpu::int8_gemm_pack_b -- how to handle them?

Reply:
same as earlier if_then_else.

Reply:
done.